ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml

```diff
@@ -144,16 +144,23 @@ let export_import_vdi rpc session_id ~exportformat sR vdi =
 let export_import_raw = export_import_vdi ~exportformat:"raw"
 let export_import_vhd = export_import_vdi ~exportformat:"vhd"
 
+module F = Storage_test.Sr_filter
+let f = F.(allowed_operations [`vdi_create; `vdi_destroy] ||> not_iso)
+
 let data_integrity_tests session_id vdi_op op_name =
-  let module F = Storage_test.Sr_filter in
-  let filter = F.(allowed_operations [`vdi_create; `vdi_destroy] ||> not_iso) in
-  [ op_name^": empty VDI", `Slow, check_vdi_unchanged ~prepare_vdi:(fun _vdi -> ()) ~vdi_op, filter
-  ; op_name^": random VDI", `Slow, check_vdi_unchanged ~prepare_vdi:(write_random_data session_id) ~vdi_op, filter
-  ; op_name^": full VDI", `Slow, check_vdi_unchanged ~prepare_vdi:(fill session_id) ~vdi_op, filter
+  [ op_name^": small empty VDI", `Slow, check_vdi_unchanged ~vdi_size:Sizes.(4L ** mib) ~prepare_vdi:(fun _vdi -> ()) ~vdi_op, f
+  ; op_name^": small random VDI", `Slow, check_vdi_unchanged ~vdi_size:Sizes.(4L ** mib) ~prepare_vdi:(write_random_data session_id) ~vdi_op, f
+  ; op_name^": small full VDI", `Slow, check_vdi_unchanged ~vdi_size:Sizes.(4L ** mib) ~prepare_vdi:(fill session_id) ~vdi_op, f
+  ]
+
+let large_data_integrity_tests session_id vdi_op op_name =
+  [ op_name^": ~2GiB empty VDI", `Slow, check_vdi_unchanged ~vdi_size:Sizes.(2L**gib) ~prepare_vdi:(fun _vdi -> ()) ~vdi_op, F.(f ||> thin_pro)
+  ; op_name^": ~2GiB random VDI", `Slow, check_vdi_unchanged ~vdi_size:Sizes.(2L**gib +* 16L**mib) ~prepare_vdi:(write_random_data session_id) ~vdi_op, F.(f ||> thin_pro)
```

Comment:
Would it make sense to use a random size here >2GiB? i.e. 2gib + random between 0 and 16mib?
Might catch a few more interesting cases where the last chunk is a partial one.

Reply:
Good idea, I've implemented that!